src/main/java/net/boomerangplatform/service/crud/TeamServiceImpl.java

```diff
@@ -449,61 +482,34 @@ public WorkflowQuotas resetTeamQuotas(String teamId) {
     }
     return flowWorkflowActivityService.findbyWorkflowIdsAndStatus(workflowIds, FlowTaskStatus.inProgress);
   }
-  
-  private WorkflowQuotas setWorkflowQuotasValues(List<WorkflowSummary> workflows,
-      List<FlowWorkflowActivityEntity> concurrentActivities,
-      Page<FlowWorkflowActivityEntity> activitiesMonthly) {
-    WorkflowQuotas workflowQuotas = new WorkflowQuotas();
-    workflowQuotas.setMaxWorkflowCount(maxWorkflowCount);
-    workflowQuotas.setMaxWorkflowExecutionMonthly(maxWorkflowExecutionMonthly);
-    workflowQuotas.setMaxWorkflowStorage(maxWorkflowStorage);
-    workflowQuotas.setMaxWorkflowExecutionTime(maxWorkflowExecutionTime);
-    workflowQuotas.setMaxConcurrentWorkflows(maxConcurrentWorkflows);
-    workflowQuotas.setCurrentWorkflowCount(workflows.size());
-    workflowQuotas.setCurrentConcurrentWorkflows(concurrentActivities.size());
-    workflowQuotas.setCurrentWorkflowExecutionMonthly(activitiesMonthly.getContent().size());
-    Integer currentWorkflowsPersistentStorage = 0;
-    for(WorkflowSummary workflow : workflows) {
-      if(workflow.isEnablePersistentStorage()) {
-        currentWorkflowsPersistentStorage += 1;
-      }
-    }
-    workflowQuotas.setCurrentWorkflowsPersistentStorage(currentWorkflowsPersistentStorage);
-    Calendar nextMonth = Calendar.getInstance();
-    nextMonth.add(Calendar.MONTH, 1);
-    nextMonth.set(Calendar.DAY_OF_MONTH, 1);
-    nextMonth.set(Calendar.HOUR_OF_DAY, 0);
-    nextMonth.set(Calendar.MINUTE, 0);
-    nextMonth.set(Calendar.SECOND, 0);
-    nextMonth.set(Calendar.MILLISECOND, 0);
-    workflowQuotas.setMonthlyResetDate(nextMonth.getTime());
-    return workflowQuotas;
-  }
 
   @Override
   public Quotas updateTeamQuotas(String teamId, Quotas quotas) {
     FlowTeamEntity team = flowTeamService.findById(teamId);
     
+    Quotas updatedQuotas = new Quotas();
+    
     if(team.getQuotas() == null) {
-      team.setQuotas(new Quotas());
+      team.setQuotas(updatedQuotas);
     }
```

Comment:
instead of declaring Quotas updatedQuotas = new Quotas(); null check team quotas. If null initialize

```
if(team.getQuotas() == null) {
team.setQuotas(new Quotas());
    }	    
```

Then use quotas on team instead of updatedQuotas